mealie/services/event_bus_service/event_bus_service.py

```diff
@@ -59,10 +72,18 @@ def dispatch(
             document_data=document_data,
         )
 
-        self.bg.add_task(self.publish_event, event=event)
+        if self.bg:
+            self.bg.add_task(self.publish_event, event=event)
+
+        else:
+            self.publish_event(event)
 
     def publish_event(self, event: Event) -> None:
         """Publishes the event to all listeners"""
         for listener in self.listeners:
             if subscribers := listener.get_subscribers(event):
                 listener.publish_to_subscribers(event, subscribers)
+
+    @staticmethod
+    def create(bg: BackgroundTasks, session=Depends(generate_session), group_id: UUID4 | None = None):
+        return EventBusService(bg, session, group_id)
```

Comment:
This should be a classmethod

Reply:
TIL how class methods work; they've been a bit of a mystery to me. Thanks!